include/cutlass/gemm/kernel/xe_gemm.hpp

```diff
@@ -243,22 +235,26 @@ class GemmUniversal<
     // Get the appropriate blocks for this sub_group -- potential for sub_group locality
     int thread_idx = int(ThreadIdxX());
     auto blk_shape = TileShape{};
-    #ifdef CUTLASS_SYCL_SWITCH_WG
-    auto m_coord = BlockIdxX();
-    auto n_coord = BlockIdxY();
-    #else
     auto m_coord = BlockIdxY();
     auto n_coord = BlockIdxX();
-    #endif
     auto l_coord = BlockIdxZ();
 
     auto blk_coord_mnkl = make_coord(m_coord, n_coord, _, l_coord);
-    int sub_group_id = thread_idx / SubgroupSize;
     constexpr auto workgroup_shape = WorkgroupTileShape{};                                                  // (SUB_M,SUB_N,SUB_K)
     constexpr auto subgroup_shape = SubgroupTileShape{};                   
 
-    auto gA = local_tile(params.mA_mk, blk_shape, take<0, 3>(blk_coord_mnkl), Step<_1,  X, _1>{});
-    auto gB = local_tile(params.mB_nk, blk_shape, take<0, 3>(blk_coord_mnkl), Step< X, _1, _1>{});
+    Tensor mA_mkl = params.mainloop.copy_A.get_pvc_tensor(make_shape(M,K,L));   //(m,k,l)
+    Tensor mB_nkl = params.mainloop.copy_B.get_pvc_tensor(make_shape(N,K,L));   //(n,k,l)
+
+    Tensor mA_mk = mA_mkl(_,_,l_coord);                                          // (m,k)
+    Tensor mB_nk = mB_nkl(_,_,l_coord);                                          // (n,k)
+
+    auto gA_mk = local_tile(mA_mk, blk_shape, make_coord(_,_,_), Step<_1,  X, _1>{});
+    auto gB_nk = local_tile(mB_nk, blk_shape, make_coord(_,_,_), Step< X, _1, _1>{});
+
+    // Slice with m_coord and n_coord
+    Tensor gA = gA_mk(_,_,m_coord,_);                                                       // (BLK_M,BLK_K,k)
+    Tensor gB = gB_nk(_,_,n_coord,_);                                                       // (BLK_N,BLK_K,k)
```

Comment:
Same here I think it should be possible to say:
```c++
Tensor gA = ocal_tile(mA_mk, blk_shape, make_coord(m_coord,_,l_coord), Step<_1,  X, _1>{});                                          
Tensor gB = local_tile(mB_nk, blk_shape, make_coord(n_coord,_,l_coord), Step< X, _1, _1>{});
```

Reply:
done something similar